rules_bot.py

```diff
@@ -161,6 +162,11 @@ def main() -> None:
     # Tag hints - works with regex
     application.add_handler(MessageHandler(TagHintFilter(), tag_hint))
 
+    # Check if user shared a bot's token
+    application.add_handler(
+        MessageHandler(filters.Regex(r"[0-9]{9}:[a-zA-Z0-9_-]{35}"), token_warning)
```

Comment:
`fiters.Regex` uses `re.search` instead of `re.match`. This means:

* if the token is not at the start of the message, this won't match
* only one token can be found

Maybe you can write a custom filter that uses `re.match`? the source code of `filters.Regex` can help as inspiration :)

Reply:
As in, manually regex the text on the `Filters.TEXT` callback, or making a PR on PTB for a new `Filters` type (or optional argument on the existing one)?

Reply:
Either use `filters.TEXT` and manually regex in the callback or create a `class MatchRegex(MessageFilter)` (or sim. naming) in `utils.py`. A PR to PTB is not necessary at this point IMO.